### PR TITLE
Feature/tao 9126/mergeBack with interactive prompt

### PR DESCRIPTION
**Pull request by oatymart**
https://oat-sa.atlassian.net/browse/TAO-9126

Added 2 methods:
- `release.mergeBack()` (expanded to handle merge conflicts)
- `release.promptToResolveConflicts()`

Tests:
`npm run test:unit`

The easiest way to truly test the positive case is to run a `taoRelease oldWayRelease` on the taoFoobar extension.

The integration test which gives better coverage to the whole tool will be added in a separate PR. I just left the empty folder and npm command here for now.

### CODE REVIEW

File: src/release.js [hunk 1]
```diff
@@ -297,9 +297,31 @@ module.exports = function taoExtensionReleaseFactory(params = {}) {
         async mergeBack() {
             log.doing(`Merging back ${releaseBranch} into ${baseBranch}`);
```
[the remark this answers is not in the document]
Comment (oatymart) [reply src/release.js]: Good catch! This was missed in the spike. Indeed, the most consistent way to have a default is like you say.

File: package.json [hunk 2]
```diff
@@ -53,8 +55,10 @@
     "nyc": "^14.1.1",
     "open-cli": "^5.0.0",
     "proxyquire": "^2.0.0",
+    "replace-in-file": "^4.1.3",
     "sinon": "^4.4.5",
     "tap-diff": "^0.1.1",
-    "tape": "^4.9.0"
+    "tape": "^4.9.0",
+    "tape-promise": "^4.0.0"
```
[the remark this answers is not in the document]
Comment (oatymart) [reply package.json]: It was an experiment, removed now.

File: .gitignore [hunk 3]
```diff
@@ -58,3 +58,5 @@ typings/
 .env
 
 .nyc_output
+
+tests/integration/temp
```
[the remark this answers is not in the document]
Comment (oatymart) [reply .gitignore]: It's for the future, but I'll move it into another PR.

File: src/release.js [hunk 4]
```diff
@@ -308,9 +308,31 @@ module.exports = function taoExtensionReleaseFactory(params = {}) {
...
 
-            await gitClient.mergeBack(baseBranch, releaseBranch);
+            try {
+                await gitClient.mergeBack(baseBranch, releaseBranch);
+                log.done();
+            }
+            catch (err) {
+                if (err && err.message && err.message.startsWith('CONFLICTS:')) {
+                    log.error(`There were conflicts preventing the merge of ${releaseBranch} back into ${baseBranch}.`);
+                    log.warn(`Please resolve the conflicts and complete the merge manually (including making the merge commit).`);
```
Comment (ricproenca-tao) [inline src/release.js]: You can use `'` here.

Comment (oatymart) [reply src/release.js]: Changed

**Pull request merged**